src/Compilers/Core/Portable/StrongName/StrongNameFileSystem.cs

```diff
@@ -1,63 +1,43 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
 using System.Diagnostics;
 using System.IO;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis
 {
+    /// <summary>
+    /// This is an abstraction over the file system which allows for us to do more thorough unit testing.
+    /// </summary>
     internal class StrongNameFileSystem
     {
         internal readonly static StrongNameFileSystem Instance = new StrongNameFileSystem();
-        protected StrongNameFileSystem()
+        internal readonly string _tempPath;
+
+        internal StrongNameFileSystem(string tempPath = null)
         {
+            _tempPath = tempPath;
         }
 
-        internal virtual byte[] ReadAllBytes(string fullPath)
+        internal virtual FileStream CreateFileStream(string filePath, FileMode fileMode, FileAccess fileAccess, FileShare fileShare)
         {
-            Debug.Assert(PathUtilities.IsAbsolute(fullPath));
-            return File.ReadAllBytes(fullPath);
+            return new FileStream(filePath, fileMode, fileAccess, fileShare);
         }
 
-        /// <summary>
-        /// Resolves assembly strong name key file path.
-        /// </summary>
-        /// <returns>Normalized key file path or null if not found.</returns>
-        internal string ResolveStrongNameKeyFile(string path, ImmutableArray<string> keyFileSearchPaths)
+        internal virtual byte[] ReadAllBytes(string fullPath)
         {
-            // Dev11: key path is simply appended to the search paths, even if it starts with the current (parent) directory ("." or "..").
-            // This is different from PathUtilities.ResolveRelativePath.
-
-            if (PathUtilities.IsAbsolute(path))
-            {
-                if (FileExists(path))
-                {
-                    return FileUtilities.TryNormalizeAbsolutePath(path);
-                }
-
-                return path;
-            }
-
-            foreach (var searchPath in keyFileSearchPaths)
-            {
-                string combinedPath = PathUtilities.CombineAbsoluteAndRelativePaths(searchPath, path);
-
-                Debug.Assert(combinedPath == null || PathUtilities.IsAbsolute(combinedPath));
-
-                if (FileExists(combinedPath))
-                {
-                    return FileUtilities.TryNormalizeAbsolutePath(combinedPath);
-                }
-            }
-
-            return null;
+            Debug.Assert(PathUtilities.IsAbsolute(fullPath));
+            return File.ReadAllBytes(fullPath);
         }
 
         internal virtual bool FileExists(string fullPath)
         {
             Debug.Assert(fullPath == null || PathUtilities.IsAbsolute(fullPath));
             return File.Exists(fullPath);
         }
+
+        internal virtual string GetTempPath() => _tempPath ?? Path.GetTempPath();
```

Comment:
Could there be a possibility that `GetTempPath()` is called more than once? If so, we might need to cache the result in `_tempPath`.

Reply:
It will be called more than once but you don't want to cache. It's a non-constant value hence you want to see the evolving value. 

---
In reply to: [245412602](https://github.com/dotnet/roslyn/pull/31986#discussion_r245412602) [](ancestors = 245412602)

Reply:
Here is a PR + Issue that shows that `GetTempPath` changing during a build cycle is something we very much observe in the wild. [https://github.com/dotnet/roslyn/pull/14417](https://github.com/dotnet/roslyn/pull/14417)

---
In reply to: [245417309](https://github.com/dotnet/roslyn/pull/31986#discussion_r245417309) [](ancestors = 245417309,245412602)